File: 18-06-2025_Divide Array Into Arrays With Max Difference.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

class Solution {
    public List<List<Integer>> divideArray(int[] nums, int k) {
        List<List<Integer>> ans = new ArrayList<>();
        int n = nums.length;
        Arrays.sort(nums);
        for (int i = 0; i < n; i += 3) {
            if (nums[i + 2] - nums[i] > k) {
                return new ArrayList<>();
            }
            ans.add(Arrays.asList(nums[i], nums[i + 1], nums[i + 2]));
        }
        return ans;
    }
}

File: 18-06-2025_Divide Array Into Arrays With Max Difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        ans = []
        nums.sort()
        n = len(nums)
        for i in range(0, n, 3):
            if nums[i+2] - nums[i] > k:
                return []
            ans.append([nums[i], nums[i+1], nums[i+2]])
        return ans